models/productDao.js

```diff
@@ -27,6 +27,39 @@ const showProducts = async (categoryId, limit, offset) => {
   }
 };
 
+const showProductDetail = async (productId) => {
+  try {
+    return await appDataSource.query(
+      `SELECT
+          products.id AS productId,
+          products.name AS productName,
+          categories.id AS categoryId,
+          categories.name AS categoryName,
+          products.description AS description,
+          products.stock,
+          products.price,
+          products.thumbnail,
+          JSON_ARRAYAGG(
+            JSON_OBJECT(
+              "productImageId", product_images.id,
+              "productImageUrl", product_images.url
+            )
+          )AS images
+        FROM products
+        LEFT JOIN product_images ON product_images.product_id = products.id
+        JOIN categories ON categories.id = products.category_id
+        WHERE products.id = ?
+      `,
+      [productId]
+    );
+  } catch (err) {
+    const error = new Error("INVALID_DATA_INPUT");
+    error.statusCode = 500;
```

Comment:
500번대 status code는 예외처리가 안 되었다는 것을 의미 하므로 400번대 status code로 수정해 주세요.
```suggestion
    error.statusCode = 400;
```